libminifi/include/utils/MinifiConcurrentQueue.h

```diff
@@ -135,6 +143,22 @@ class ConcurrentQueue {
   std::deque<T> queue_;
 };
 
+// Enables batched operations on a ConcurrentQueue by holding a reference to the queue and locking it's mutex
+// until it leaves the scope. Can help performance by avoiding multiple locking in loops, etc.
+template <typename T>
+class LockedConcurrentQueue {
+ public:
+  explicit LockedConcurrentQueue(ConcurrentQueue<T>& concurrentQueue)
+      : queue_(std::ref(concurrentQueue.queue_))
+      , lock_(concurrentQueue.mtx_) {
+  }
+
+  std::deque<T>* operator->() const { return &queue_.get(); }
```

Comment:
This exposes the underlying queue object. This is a big no-no.

Reply:
It has already been discussed and if everyone agrees with this approach I'm totally fine with providing a public API instead of overloaded operators.